test/BondFactory.spec.ts

```diff
@@ -64,6 +90,87 @@ describe("BondFactory", async () => {
       await expect(createBond(factory)).to.emit(factory, "BondCreated");
     });
 
+    it("should revert on less collateral than convertible ratio", async () => {
+      await factory.grantRole(ISSUER_ROLE, owner.address);
+      await expect(
+        createBond(factory, {
+          collateralRatio: utils.parseUnits(".25", 18),
+          convertibleRatio: utils.parseUnits(".5", 18),
+        })
+      ).to.be.revertedWith("CollateralRatioLessThanConvertibleRatio");
+    });
+
+    it("should revert on too big of a token", async () => {
+      const { paymentToken: bigPaymentToken } = await (
+        await tokenFixture([20])
+      ).tokens[1];
+
+      await factory.grantRole(ISSUER_ROLE, owner.address);
+      await expect(
+        createBond(factory, { paymentToken: bigPaymentToken.address })
+      ).to.be.revertedWith("DecimalsOver18()");
+    });
+
+    describe("Should revert on invalid maturity date", async () => {
```

Comment:
is there convention with uppercase Should to describe sub-tests? my first thought would be something like
`describe("invalid maturity dates")`